robusta_krr/strategies/simple.py

```diff
@@ -48,11 +48,24 @@ class SimpleStrategy(BaseStrategy[SimpleStrategySettings]):
     __display_name__ = "simple"
     __rich_console__ = True
 
-    def run(self, history_data: HistoryData, object_data: K8sObjectData) -> RunResult:
+    def __calculate_cpu_proposal(self, history_data: HistoryData, object_data: K8sObjectData) -> ResourceRecommendation:
+        if object_data.hpa is not None and object_data.hpa.target_cpu_utilization_percentage is not None:
+            return ResourceRecommendation.undefined()
+
         cpu_usage = self.settings.calculate_cpu_proposal(history_data[ResourceType.CPU].data)
+        return ResourceRecommendation(request=cpu_usage, limit=None)
+
+    def __calculate_memory_proposal(
+        self, history_data: HistoryData, object_data: K8sObjectData
+    ) -> ResourceRecommendation:
+        if object_data.hpa is not None and object_data.hpa.target_memory_utilization_percentage is not None:
```

Comment:
I think this might be a bug.
For an HPA that is scaled down all the way we will still run memory recommendations
<img width="309" alt="Screen Shot 2023-06-22 at 15 42 57" src="https://github.com/robusta-dev/krr/assets/97387909/b0a750ac-8625-4dbd-b69e-e3d6e1ef9e0a">

Reply:
I don't understand. In your screenshot averageUtilization is `0`, not `null`. So we will not run the calculation

Reply:
cpu averageUtilization is 0 not memory,
see how your function __get_metric(hpa, "memory") would run on this

Reply:
Recommendation for memory is expected when there is only hpa for CPU and vice-verca